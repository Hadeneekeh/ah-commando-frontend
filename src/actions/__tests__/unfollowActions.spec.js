import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import nock from 'nock';
import {
  unFollowUser,
} from '@Actions/unfollowActions';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const url = 'https://a-haven-staging.herokuapp.com/api/v1';

describe('testing redux actions', () => {
  let store;
  const response = {
    data: {
      profile: {},
    },
  };

  const error = {
    response: {
      data: {},
    },
  };
  beforeEach(() => {
    store = mockStore({});
  });

  afterEach(() => {
    nock.cleanAll();
  });

  it('Should follow a user successfully', async () => {
    const username = 'martins';
    nock(url)
      .delete(`/profiles/${username}/follow`)
      .reply(200, response);

    return store.dispatch(unFollowUser(username))
      .then(() => {
        expect(store.getActions()).toMatchSnapshot();
      });
  });

  it('Should error as expected', async () => {
    const username = 'martins';
    nock(url)
      .delete(`/profiles/${username}/follow`)
      .reply(400, error);

    return store.dispatch(unFollowUser(username))
      .then(() => {
        expect(store.getActions()).toMatchSnapshot();
      });
  });
});
