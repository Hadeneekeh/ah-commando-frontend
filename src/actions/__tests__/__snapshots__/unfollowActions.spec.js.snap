// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing redux actions Should error as expected 1`] = `
Array [
  Object {
    "type": "UNFOLLOW_USER_START",
  },
  Object {
    "error": Object {
      "response": Object {
        "data": Object {},
      },
    },
    "type": "UNFOLLOW_USER_FAILURE",
  },
]
`;

exports[`testing redux actions Should follow a user successfully 1`] = `
Array [
  Object {
    "type": "UNFOLLOW_USER_START",
  },
  Object {
    "payload": undefined,
    "type": "UNFOLLOW_USER_SUCCESS",
  },
]
`;
